### PR TITLE
RESTWS-808: utility classes should not have public constructors

### PR DESCRIPTION
**Pull request by theanandankit**
## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-808

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by HerbertYiga** [comment]
Thanks cc @ibacher

**Comment by dkayiwa** [comment]
@HerbertYiga why are you copying in @ibacher?

File: omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderUtil.java [hunk 1]
```diff
@@ -22,6 +22,10 @@
 import org.openmrs.api.context.Context;
 
 public class OrderUtil {
+
+	private OrderUtil(){
+		throw new IllegalStateException("Utility class");
```
Comment (sacull) [inline omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderUtil.java:27]: Interesting idea, I didn't think of it myself. The build process runs without any warnings or errors? (unfortunately I can't compile it on the computer I'm working on at the moment)

Comment (theanandankit) [reply omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderUtil.java:27]: @sacull worked fine in my computer and i followed the resource provided by you https://sonar.openmrs.org/coding_rules#rule_key=squid%3AS1118

**Review by sacull — APPROVED** [review]
It looks good to me. ;)

**Comment by dkayiwa** [comment]
If you are a developer and get an exception with message `Utility class`, would that be meaningful enough to know what has gone wrong?

**Comment by sacull** [comment]
> If you are a developer and get an exception with message `Utility class`, would that be meaningful enough to know what has gone wrong?

@theanandankit There is quite a point in this, I hadn't thought about it before. Now I think that this message should actually be improved. Maybe something like "Unable to instantiate utility class"? Or maybe there's something better that you can see and I can't. ;)

**Comment by ibacher** [comment]
Personally, I think adding the exception is a bit unnecessary overkill. Just marking the constructor private should be sufficient.

**Comment by dkayiwa** [comment]
I agree with @ibacher 😊

**Comment by theanandankit** [comment]
@ibacher @dkayiwa @sacull thank you for your valuable comments. I just changed the constructor to private empty constructor.

**Pull request merged**